### PR TITLE
[ENHANCEMENT] Polish the dark mode; Overhaul submit form; Fix dark mode skeleton components;

### LINKED ISSUE #64: Add Dark Theme for Shimmer
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Currently we don't have dark theme for Shimmer UI.

### Expected Behavior

We need to add dark theme Support for the Shimmer UI.

### Reproduction URL

https://wanderlust-beta.vercel.app/

### Reproduction steps

1. First Open URL.
2. Then Switch System to dark theme .
3. Then if you reload website.
4. Then before posts are loaded 
5. See shimmer cards render which are white.

### Screenshots

dark

### Logs

_No response_

### Browsers

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by chinmaykunkikar**
## Summary
Fix dark mode skeleton components; Polish the dark mode; Overhaul submit form;

## Description

1. **Dark Color Variant:**
   - Introduced `dark` color variant in Tailwind CSS configuration for unambiguous color assignment.

2. **Light Color Variant:**
   - Added `light` variant in Tailwind CSS, expanding color choices for enhanced design flexibility.

3. **Blog Form Overhaul:**
   - Revamped blog creation form for accessibility, increased whitespace, refined placeholders, and improved width.

4. **Unified Color Scheme:**
   - Implemented `*-slate-*` universally, ensuring consistent and hierarchical text colors.

5. **Background Color Adjustment:**
   - Replaced light mode's white background with `slate-50` for aesthetic alignment with UI practices.

6. **Homepage Design Improvements:**
   - Enhanced visual hierarchy across homepage sections for optimized user experience.

7. **Text Formatting and Styling:**
   - Converted title case text to sentence case, aligning with Google's writing practices.

8. **Button Interactivity:**
   - Added hover state to Create Post button for improved user feedback.

9. **Skeleton Loading Enhancement:**
   - Optimized post skeleton loading for smoother and visually appealing experience.

10. **Font Configuration:**
    - Integrated `Poppins` font directly into Tailwind CSS, eliminating arbitrary value classes.

11. **Category Pill Component Updates:**
    - Introduced `<CategoryPill />` component, replacing `util/category-props.tsx` in line with React principles.
    - Added dark mode support with darker light colors for visibility.
    - Adjusted CategoryPill colors from `bg-*-100` to `bg-*-200` for enhanced prominence.
    - Set CategoryPill font to medium for coherence with Figma design.

## Images

<details>
<summary><h3>Headers</h3></summary>

**Before**

wanderlust-beta vercel app_

**After**

wanderlust-pp9y1mnbe-krishnaacharya-official vercel app_

</details>
<details>
<summary><h3>Skeletons (Dark)</h3></summary>

**Before**

wanderlust-beta vercel app_ (4)

**After**

localhost_5173_

</details>
<details>
<summary><h3>Featured and latest posts (Dark)</h3></summary>

**Before**

wanderlust-beta vercel app_ (1)

**After**

wanderlust-pp9y1mnbe-krishnaacharya-official vercel app_ (1)

</details>
<details>
<summary><h3>Featured and latest posts (Light)</h3></summary>

**Before**

wanderlust-beta vercel app_ (5)

**After**

wanderlust-pp9y1mnbe-krishnaacharya-official vercel app_ (4)

</details>
<details>
<summary><h3>All blog posts cards (Dark)</h3></summary>

**Before**

wanderlust-beta vercel app_ (2)

**After**

wanderlust-pp9y1mnbe-krishnaacharya-official vercel app_ (2)

</details>
<details>
<summary><h3>Create blog form (Dark)</h3></summary>

**Before**

wanderlust-beta vercel app_ (3)

**After**

wanderlust-pp9y1mnbe-krishnaacharya-official vercel app_ (3)

</details>
<details>
<summary><h3>Create blog form (Light)</h3></summary>

**Before**

wanderlust-beta vercel app_ (6)

**After**

wanderlust-pp9y1mnbe-krishnaacharya-official vercel app_ (5)

</details>
<details>
<summary><h3>Details page (Dark)</h3></summary>

**Before**
wanderlust-beta vercel app_ (7)

**After**
localhost_5173_details-page

</details>
<details>
<summary><h3>Details Page (Light)</h3></summary>

**Before**
wanderlust-beta vercel app_ (8)

**After**
localhost_5173_details-page (1)

</details>
<details>
<summary><h3>Modal</h3></summary>

**Before**

wanderlust-beta vercel app_add-blog

**After**

wanderlust-ba6yxpzgy-krishnaacharya-official vercel app_add-blog

</details>

## Issue(s) Addressed
- Closes #64
- Related to #62 

## Prerequisites

- [x] I have followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)

### CODE REVIEW

File: frontend/tailwind.config.js
Comment (chinmaykunkikar) [inline frontend/tailwind.config.js:17]: @SankalpHaritash21 

The reason why I am using this syntax is because I want to access the official Tailwind colors. The Tailwind library offers convenient JavaScript color helper utilities through `tailwindcss/colors`. This utility provides an object with hex codes for all official Tailwind colors.

<details>
<summary>If you put a <code>console.log(colors)</code> somewhere, the output will resemble this: <i>(expand this block)</i></summary>
<br />

```js
// Output of console.log(colors)
{
  inherit: 'inherit',
  current: 'currentColor',
  transparent: 'transparent',
  black: '#000',
  white: '#fff',
  slate: {
    '50': '#f8fafc',
    '100': '#f1f5f9',
    '200': '#e2e8f0',
    '300': '#cbd5e1',
    '400': '#94a3b8',
    '500': '#64748b',
    '600': '#475569',
    '700': '#334155',
    '800': '#1e293b',
    '900': '#0f172a',
    '950': '#020617'
  },
  gray: {
    '50': '#f9fafb',
    '100': '#f3f4f6',
    ...
  },
  stone: {
    ...
  },
  red: {
    ...
  },
 ...
}
```

</details>

This structure is essentially an object of objects, where colors (`slate`, `red`, `cyan`, etc.) act as child keys, and their corresponding values are objects with keys represented as strings (`'50'`, `'100'`, `'200'`,...). Each of these keys have their values as different shades of the colors (`'#f8fafc'`, `'#f1f5f9'`, `'#e2e8f0'`,...).

To get a specific shade, we use the standard JavaScript object property access syntax: `obj1[childKey][key]` or **`obj1.childKey[key]`** or `obj1.childKey.key`.

For example, I'm using `colors.slate[500]` to fetch the `500` shade of the `slate` color.

Comment (SankalpHaritash21) [reply frontend/tailwind.config.js:17]: @chinmaykunkikar  have u installed tw/colors or it is inside by default

Comment (SankalpHaritash21) [reply frontend/tailwind.config.js:17]: @chinmaykunkikar now I find where I failed I tryed to use
 DEFAULT: "colors.slate" it as a string

Comment (SankalpHaritash21) [reply frontend/tailwind.config.js:17]: Thanku for this

Comment (chinmaykunkikar) [reply frontend/tailwind.config.js:17]: @SankalpHaritash21 it comes installed by default if you're using tailwind in your project.
And, IIRC you used `colors.slate` without specifying a shade (500, 300, etc).

**Comment by krishnaacharyaa** [comment]
@chinmaykunkikar, thank you for this wonderful PR, Hats off to your patience. Now, please spare me some good amount of time to digest all the concepts which are proposed in this PR ;)

**Comment by chinmaykunkikar** [comment]
@krishnaacharyaa I'll update the description with screenshots and I'll mark this PR ready for review :)

**Comment by chinmaykunkikar** [comment]
@krishnaacharyaa whenever you're ready :)

**Comment by SankalpHaritash21** [comment]
@chinmaykunkikar  actually I am also tracking ur all commit.

and same as @krishnaacharyaa https://github.com/krishnaacharyaa/wanderlust/pull/67#issuecomment-1837039569
I am also trying to understand as ur approach as is new to me .

**Comment by krishnaacharyaa** [comment]
@chinmaykunkikar, I wouldn't hesitate to say it again, Thanks for your patience in writing such a beautiful PR description, I am just in love with that, Google docs which you shared did really help you ;) Looking forward to read the same very soon, and ofcourse all the details that are addressed, really crisp to the point, it just enhances the quality exponentially. Hands down for combining what could be ideally task of potentially 3 to 4 PRs.

Just couple of suggestions/opinions I would like to bring up:
1. 
image
2. 
image

For some reason, I'm not convinced with the alignment, Can we actually get the alignment to the heading as well? I understand people and may be google even suggests this, but I am not big fan of this style personally I feel something is misaligned, Can you kindly convince me in couple of sentences why this would be best for our project, and would there be any harm for us to use the aligned title and description, What's your thoughts on this? Haven't even you felt this? (at this point I am curious)

**Comment by chinmaykunkikar** [comment]
> Can we actually get the alignment to the heading as well?

I didn't get you. Are you saying you don't like the center alignment? Or do you want it to be full screen width?

**Review by krishnaacharyaa — COMMENTED** [review]
Thank you for the PR, Kindly address the review comments.

**Comment by krishnaacharyaa** [comment]
> > Can we actually get the alignment to the heading as well?
> 
> > and would there be any harm for us to use the aligned title and description
> 
> @krishnaacharyaa I didn't get you. Are you saying you don't like the center alignment? Or do you want it to be full screen width?

Meaning, Title is forced to the left and the description is centre aligned, I am proposing can we keep it uniform to both, either stretched for both title and description or centre for both.

**Comment by krishnaacharyaa** [comment]
> Updated: As it is not aligned pick img

Hey @SankalpHaritash21, Thank you for your valuable insight, Couple of points to add, @chinmaykunkikar we might have to even work on the padding in the mobile screen (We can go with 16px both the ends in mobile), and the responsive font size as well, similar to home page.

File: frontend/src/components/category-pill.tsx
Comment (chinmaykunkikar) [reply frontend/src/components/category-pill.tsx:10]: Ah, I forgot about that cheeky `any` while I was trying something out. Apologies, and thanks for pointing that out.
To answer your concern, I'm afraid we can't change that to a simple `string` type to the object that has the categories as their keys . We'll have to define a type something like:

```js
type categoryColorsType = {
    Travel: string,
    Mountains: string,
    ...
}
```

This seems redundant to me. We'll have to keep editing the type if we wish to change the categories.

Let's figure out a different way for this.

File: frontend/src/components/category-pill.tsx
Comment (chinmaykunkikar) [reply frontend/src/components/category-pill.tsx:32]: Thanks for the suggestion. I'll change that (I hate naming variables).

File: frontend/src/components/category-pill.tsx
Comment (chinmaykunkikar) [reply frontend/src/components/category-pill.tsx:19]: hmm.

File: frontend/src/pages/add-blog.tsx
Comment (chinmaykunkikar) [reply frontend/src/pages/add-blog.tsx:219]: Did that here: https://github.com/krishnaacharyaa/wanderlust/pull/67/commits/9f5a0d36b69982c57585fb2757c00da3ffcebcde. Didn't work.
I think it doesn't happen for `className` attributes. It's a hit or miss at least.
Because iirc, I had to break the line manually in a personal repo of mine.

**Comment by chinmaykunkikar** [comment]
> work on the padding in the mobile screen

Jussst changed that locally before this comment lol. I hated that one too.
It actually skipped my mind to test the form on mobile screens. I'm doing some changes in that regards right now.

File: frontend/src/components/category-pill.tsx
Comment (krishnaacharyaa) [reply frontend/src/components/category-pill.tsx:10]: Oops I almost missed the key value pairs.
Can we use something like this 
`const categoryColors = new Map<string, string>([["travel", "classes"], ["mountains","classes"]])`
 we would have to then stringify the category when needed

File: frontend/src/pages/add-blog.tsx
Comment (krishnaacharyaa) [reply frontend/src/pages/add-blog.tsx:219]: Yeah, i can even see it in `frontend/tsconfig.json`, can we do something for that? 

p.s : https://stackoverflow.com/questions/71377814/prettier-code-formatting-is-not-splitting-classnames-in-jsx-or-html got this from the net. We can use something which suits us the best

**Comment by chinmaykunkikar** [comment]
> Title is forced to the left and the description is centre aligned

I see.
2. For the form page, I wanted to keep the arrow to the top left corner and move the "Create blog" along with the form but I didn't touch it because I thought it's a part of the navigation. That one bugged me as well.

So, Do you want me to move the text, keeping the nav arrow in place?

---

Hmm. for 1, I'll tell you my thought behind shrinking the width, and moving the blog body (description) to the center of the screen.
**Limited width** - Since blogs are text heavy pages, having longtext lines are not really ideal for reading content. If you start reading a line from one end to the other end of your screen, you'll have to adjust your eyes to go to the next line, essentially breaking your reading flow. There are UX case studies behind it, and you'll also see most text-heavy sites with a limited width, that is (mostly) aligned to the center of the screen, with A LOT of whitespace to the either side of the text.

medium com_transit-map-central_tutorial-creating-a-simple-schematic-transit-map-using-illustrator-5d46d9fa3885
_(Medium has cracked this code long ago)_


**Not aligning with the heading**: It would've been okay to align the text to the start here, but it'd have felt odd seeing a huge whitespace to the right side of the user's screen (because of the shrinking). And I never thought about moving the title to the center with the rest of the text, because I always thought it was disconnected (in terms of style) from the body (description) because of the background image behind the title.

Do you want me to move the title to the center?

File: frontend/src/components/blog-feed.tsx
Comment (chinmaykunkikar) [reply frontend/src/components/blog-feed.tsx:79]: Yep!

**Comment by krishnaacharyaa** [comment]
> didn't touch it because I thought it's a part of the navigation

No its not.

> So, Do you want me to move the text, keeping the nav arrow in place?

Yes please.

> And I never thought about moving the title to the center with the rest of the text, because I always thought it was disconnected (in terms of style) from the body (description) because of the background image behind the title.

UX part I agree with that, and i have no problem, even i had the same intention initially building the project, because of some PR, it got slipped from me, so we are seeing it stretched. And the above makes sense to me, the disconnection part, I guess we will let it be the way you have done it now, I'm fine with this.

File: frontend/src/pages/add-blog.tsx
Comment (chinmaykunkikar) [reply frontend/src/pages/add-blog.tsx:219]: The Headwind plugin that's mentioned in the answer is not maintained since 2021, and it will definitely conflict with the official tailwind plugin that we're using.
I'll look into it.

Comment (krishnaacharyaa) [reply frontend/src/pages/add-blog.tsx:219]: `word-wap` will be the only option then, yeah sure, it's not of top priority

**Comment by chinmaykunkikar** [comment]
> Yes please.

Done.

> As it is not aligned pick img

Done.

File: frontend/src/components/category-pill.tsx
Comment (chinmaykunkikar) [reply frontend/src/components/category-pill.tsx]: Done.

**Comment by SankalpHaritash21** [comment]
Hi @chinmaykunkikar @krishnaacharyaa 

detail design


Can we align  title with detail and @chinmaykunkikar as your choice.
@chinmaykunkikar  Now I came to know about professional pr. I thought u r making doc for wanderlust website. It look like. @chinmaykunkikar can u tell me from where you get such experience. Plz tell me too.

**Comment by chinmaykunkikar** [comment]
> Create Blog post page looks good to me, in terms of responsiveness, the home page looks a bit missed up, if we are not getting engagements from contributors I'll take it up. No worries. Referring to #13 

@krishnaacharyaa, I didn't bother much about the homepage responsiveness.
Tomorrow, I'll skim through some code and if I think I can do it without breaking a sweat, I'll push it in this PR itself. Otherwise, please take it up.

**Comment by chinmaykunkikar** [comment]
> @chinmaykunkikar  your code look great and the approach used is good. I approved ur code.
> 
> hey @chinmaykunkikar  can u plz add some meta data in index file as It will help in better recognition in google.
> 
> updated:
> 
> 
>   <head>
>     <meta charset="UTF-8" />
>     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
>     <link
>       href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700&display=swap"
>       rel="stylesheet"
>     />
>     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
>     <title>Wanderlust</title>
>   </head>
> 

Perhaps a topic for another PR? :)

**Comment by chinmaykunkikar** [comment]
> hey @chinmaykunkikar  can u plz add some meta data in index file as It will help in better recognition in google.
> 
> updated:
> 
> ```html
>   <head>
>     <meta charset="UTF-8" />
>     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
>     <link
>       href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700&display=swap"
>       rel="stylesheet"
>     />
>     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
>     <title>Wanderlust</title>
>   </head>
> ```
> 
> Edit: add a code block.

P.S. Please learn to use code blocks when you want to show or refer some code.
For example, in this comment, if I hadn't quoted your reply, I wouldn't have known what you're trying to say.
Because this is what it looked like in your comment before editing:

**Comment by chinmaykunkikar** [comment]
> We'll take this up seperately

Sure

> And the title one even we'll do it seperately!

Yes, sure.

> If you can just do the Map type of category, that would suffice for now. If you feel it needs some dedicated time we'll take it up later.

I'll do it in this PR itself.

> p.s: this has some really nice enhancements, so I'm hurrying to merge this, hope you understand and don't mind 😊 

I get you. But... relax, this is the only task that that's pending for this PR. You'll (probably) just have to click on the merge button once I push it.
Once this is kept on hold, it will take forever to get it fixed.

**Comment by krishnaacharyaa** [comment]
> I'll do it tomorrow because I'm traveling today.
oops okay... Take your time.

File: frontend/src/utils/category-colors.ts
Comment (krishnaacharyaa) [inline frontend/src/utils/category-colors.ts:2]: please rename to `CategoryColors`, Map seems irrelevant to me, as it is type

File: frontend/src/utils/category-colors.ts
Comment (krishnaacharyaa) [inline frontend/src/utils/category-colors.ts:4]: `categoryAndColors` to `categoryColors`

**Review by krishnaacharyaa — COMMENTED** [review]
Thank you for very unique and interesting approach, looks cool to me, just some naming conventions if we can update :D

**Comment by krishnaacharyaa** [comment]
@chinmaykunkikar just couple of things, pardon me for being very picky and specific.
image
The arrow has a little padding at the left, can you fix that

And here
image
Can we have the Title in the same alignment as Back button, that feels reasonable to me, so that it is heading and has some seperation from the form itself

**Review by krishnaacharyaa — APPROVED** [review]
Thank you @chinmaykunkikar, LGTM. It's gonna be the biggest PR with 24 commits till date ;)

**Pull request merged**
